### PR TITLE
nixos/test: Use retry() in all looping functions that need timeouts

### PR DESCRIPTION
**Pull request by tfc**
###### Motivation for this change

#72828

Related to #74868. Use `retry` in all functions that should wait between retries and also timeout after some time so a test won't hang forever. These changes also resemble what the original perl driver does.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @worldofpeace @flokli

### CODE REVIEW

**Comment by jtojnar** [comment]
Maybe using the decorator from https://pypi.org/project/retry/ or https://pypi.org/project/retrying/ would be nicer.

**Comment by tfc** [comment]
> Maybe using the decorator from https://pypi.org/project/retry/ or https://pypi.org/project/retrying/ would be nicer.

On the one hand: Yes. On the other hand: an extra dependency for such a little function seems like too much. What do you think?

File: nixos/tests/pam-oath-login.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import ./make-test.nix ({ ... }:
```
Comment (flokli) [inline nixos/tests/pam-oath-login.nix]: This should go in a separate PR.

**Review by flokli — CHANGES_REQUESTED** [review]

**Comment by flokli** [comment]
@GrahamcOfBorg test pam-oath-login

File: nixos/tests/pam-oath-login.nix (hunk 1 above)
Comment (flokli) [reply nixos/tests/pam-oath-login.nix]: Well, at least it's 2 separate commits, so we can merge this as is.

Still, please open 2 separate PRs next time.

Comment (flokli) [reply nixos/tests/pam-oath-login.nix]: Well, at least it's 2 separate commits, so we can merge this as is.

Still, please open 2 separate PRs next time.
